options/m_option.h

```diff
@@ -256,6 +256,8 @@ union m_option_value {
     struct m_channels channels;
 };
 
+extern const union m_option_value m_option_value_default;
```

Comment:
Not a fan of this. Unless LTO is used it will be opaque to the compiler for no good reason. Why not use `static` in the header?

It'll be optimized out when unused. And I think mpv builds with -Wno-unused so shouldn't result in warnings either (if it does, there's always `__attribute((unused))` or similar).

Reply:
> Not a fan of this. Unless LTO is used it will be opaque to the compiler for no good reason. Why not use static in the header?

It doesn't matter in practice. Compiler will not optimize it out anyway beyond the "memcpy" equivalent, simply because this union does not fit in the register. It contains few struct in itself, which makes it quite bulky.

I had this in back of my head, but decided not to do that. Simply because it fits with `m_option_path_separator` that is already there. And any optimization is in fact not important.

But if you feel like it matters to you, should I change that?

Reply:
> It contains few struct in itself, which makes it quite bulky.

Not in front of a computer right now, how bulky are we talking about? Anything less than 64 bytes has a good chance of getting optimized via wider vector instructions.

Also, it's not *just* the initialization itself, the compiler being able to see the value makes it capable of doing other optimization based on it.

> should I change that?

I think it's better (both "optimization" wise but also readability wise as I need to lookup only one definition in the header).

But I agree that it doesn't matter a whole lot in the big picture. So feel free to make the call yourself.

Reply:
> Anything less than 64 bytes has a good chance of getting optimized via wider vector instructions.

It has 24 bytes.

> I think it's better (both "optimization" wise but also readability wise as I need to lookup only one definition in the header).

I agree, in fact I was worried people would say otherwise and complain that I add static global data in headers.